src/main/java6/net/finmath/time/ScheduleGenerator.java

```diff
@@ -127,8 +126,8 @@ private ScheduleGenerator() {
 	 * t = 0 corresponds to the reference date.
 	 * 
 	 * @param referenceDate The date which is used in the schedule to internally convert dates to doubles, i.e., the date where t=0.
-	 * @param startDate The start date of the first period.
-	 * @param maturity The end date of the first period.
+	 * @param startDate The start date of the first period (this may/should be an unadjusted date).
```

Comment:
Ups, I was confused by my own comment - lol. The intention was: startDate does not require an adjustment, adjustments take place as part of the roll out (but of course, startDate can be a businessday (and a non-businessday).

W.r.t. your second comment:

- I would not add to much enforcement - checks could happen "outside", AND
- startState has to be allowed to be a non-business day, if not it would be impossible to create the swap with periods 1.1, 1.4, 1.7, 1.10, ... (the 1.1. will be adjusted during the roll out).

(I've checked the roll out a while ago and it was consistent with other systems, - afaik).